reference.md

```diff
@@ -0,0 +1,1454 @@
+    Uploading goodsnap_version_arch.snap.
+    Uploading integration_tests/snaps/basic/u1test20160920_0.1_all.snap [                  ]   0%
+    Uploading integration_tests/snaps/basic/u1test20160920_0.1_all.snap [==================] 100%
+    Ready to release!|
+    Revision 1 of 'goodsnap' created.
+
+### release
+
+    $ snapcraft release pushedsnap 1 edge
+    The 'edge' channel is now open.
+
+    Channel    Version    Revision
+    ---------  ---------  ----------
+    stable     -          -
+    candidate  -          -
+    beta       -          -
+    edge       0.1        1
+
+### Errors
+
+Try a store command without login:
+
+    $ snapcraft register test
+    Registering test.
+    No valid credentials found. Have you run "snapcraft login"?
+    Invalid credentials: [].
+
+Try to register a snap without signing the agreement:
+
+    $ snapcraft register test
+    Registering test.
+    Expecting value: line 1 column 1 (char 0)
+
+Try to register a snap without developer name or country: todo
+
+Try to register two snaps in a short period:
+
+    $ snapcraft register second-snap
+    Registering second-snap.
+    You must wait 152 seconds before trying to register your next snap.
+
+Try to register a reserved name:
+
+    $ snapcraft register bash
+    Registering bash.
+    The name 'bash' is reserved.
+
+    If you are the publisher most users expect for 'bash' then please claim the name at 'https://myapps.developer.ubuntu.com/dev/click-apps/register-name-dispute/?series=16&name=bash'
+
+Try to register a name already registered:
+
+    $ snapcraft register already-registered
+    Registering already-registered.
+    The name 'already-registered' is already taken.
+
+    We can if needed rename snaps to ensure they match the expectations of most users. If you are the publisher most users expect for 'already-registered' then claim the name at 'https://myapps.developer.ubuntu.com/dev/click-apps/register-name-dispute/?series=16&name=u1test201607183'
+
+Try to register a name with an invalid name:
+
+    $ snapcraft register _invalid   
+    Registering _invalid.
+    Registration failed.
+
+Try to push a file that doesn’t exist:
+
+    $ snapcraft push idontexist
+    The file 'idontexist' does not exist.
+
+Try to push a file that’s not a snap:
+
+    $ snapcraft push imnotasnap
+    Uploading imnotasnap.
+    Read on filesystem failed because EOF
+    Read on filesystem failed because EOF
+    Can't find a SQUASHFS superblock on imnotasnap
+    Command '['unsquashfs', '-d', '/tmp/tmphedels1u/squashfs-root', 'imnotasnap', '-e', 'meta/snap.yaml']' returned non-zero exit status 1
+
+Try to upload a snap not registered:
+
+    $ snapcraft push notregistered_version_arch.snap
+    Uploading notregistered_version_arch.snap.
+    Uploading notregistered_version_arch.snap [                                                 ]   0%
+    Uploading notregistered_version_arch.snap [=================================================] 100%
```

Comment:
This fully uploads the files before returning the error. We need to err before we upload anything.